app/policies/hspd12_policy.rb

```diff
@@ -0,0 +1,25 @@
+class Hspd12Policy
+  def initialize(session:, user:)
+    @session = session
+    @user = user
+  end
+
+  def piv_cac_setup_required?
+    piv_cac_required? && !piv_cac_enabled?
+  end
+
+  def piv_cac_required?
+    return if session.blank? || Figaro.env.allow_piv_cac_required != 'true'
+    sp_session = session.fetch(:sp, {})
+    return if sp_session.blank?
+    sp_session[:requested_attributes]&.include?('x509_presented')
```

Comment:
@jmhooper just made a PR to have AAL3Policy stop depending directly on session --- which might mess with this when merged. Should we get ahead of that and update this to just need `requested_attributes` rather than access to the whole session? 

PR: https://github.com/18F/identity-idp/pull/3977

(the reason I comment here is that as far as I can tell in these classes, the only reason we pass around session is to access this attribute list)

Reply:
I forgot to cull this.  It's no longer needed.

Reply:
done